src/main/java/org/jabref/gui/util/Theme.java

```diff
@@ -47,31 +54,31 @@ public Theme(String path, PreferencesService preferencesService) {
         if (StringUtil.isBlank(path) || BASE_CSS.equalsIgnoreCase(path)) {
             // Light theme
             this.type = Type.LIGHT;
-            this.additionalCssToLoad = Optional.empty();
+            this.cssUrl = Optional.empty();
         } else {
-            Optional<URL> cssResource = Optional.ofNullable(JabRefFrame.class.getResource(path));
-            if (cssResource.isPresent()) {
+            URL url = JabRefFrame.class.getResource(path);
+            if (url != null) {
                 // Embedded dark theme
                 this.type = Type.DARK;
             } else {
                 // Custom theme
                 this.type = Type.CUSTOM;
-                if (Files.exists(pathToCss)) {
-                    try {
-                        cssResource = Optional.of(pathToCss.toUri().toURL());
-                    } catch (MalformedURLException e) {
-                        cssResource = Optional.empty();
-                    }
+                try {
+                    url = pathToCss.toUri().toURL();
+                } catch (MalformedURLException e) {
+                    url = null;
                 }
             }
 
-            if (cssResource.isPresent()) {
-                additionalCssToLoad = cssResource;
-                LOGGER.debug("Using css {}", path);
-            } else {
-                additionalCssToLoad = Optional.empty();
-                LOGGER.warn("Cannot load css {}", path);
-            }
+            this.cssUrl = Optional.ofNullable(url);
+        }
+    }
+
+    private Optional<URL> additionalCssToLoad() {
+        if (type == Type.CUSTOM && !Files.exists(pathToCss)) {
```

Comment:
Can you add some log for the case that the file doesn't exists?

Reply:
Good shout, I will add that logging and also for malformed URL since both cases previously were covered by some logging.

Reply:
Logging added.

I've also added a unit test for the Theme class. The installCss() methods are still uncovered: a few interact with JavaFX concrete classes and use Platform.runLater() so would need more effort to mock nicely. Also, I haven't covered error handling of malformed URLs. I hit into a funny thing, that on Windows, colons in paths are always interpreted as drive names. So file path processing fails before we get to URL handling. I think I'll raise a separate issue because this has led me to realise that Windows drive handling is another edge case to deal with.

Still, this gets us to 53% coverage of the Theme class 🎉

Reply:
Take your time, there is no rush.
We are really happy, that you take care of this issue and also care about test coverage.

Reply:
Thanks for your feedback @calixtus, should I resolve conversations or do I leave that to reviewers?

Reply:
I'd say: If a remark is resolved, it's resolved...